<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>YouTube Summarizer – Gemini AI</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    
    <!-- YouTube URL Input -->
    <item>
     <layout class="QHBoxLayout" name="urlLayout">
      <item>
       <widget class="QLabel" name="label_url">
        <property name="text">
         <string>🔗 YouTube URL:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_url">
        <property name="placeholderText">
         <string>Enter YouTube URL...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_summarize">
        <property name="text">
         <string>▶ Summarize</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Transcript Viewer -->
    <item>
     <widget class="QLabel" name="label_transcript">
      <property name="text">
       <string>📄 Transcript:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="textEdit_transcript">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>

    <!-- Summary Viewer -->
    <item>
     <widget class="QLabel" name="label_summary">
      <property name="text">
       <string>🧠 Summary (from Gemini):</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="textEdit_summary">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>

    <!-- Buttons -->
    <item>
     <layout class="QHBoxLayout" name="buttonLayout">
      <item>
       <widget class="QPushButton" name="button_save">
        <property name="text">
         <string>💾 Save Summary</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_reset">
        <property name="text">
         <string>♻️ Reset</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_exit">
        <property name="text">
         <string>❌ Exit</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Progress Bar -->
    <item>
     <widget class="QProgressBar" name="progressBar">
      <property name="visible">
       <bool>false</bool>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
     </widget>
    </item>

   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>


